bpfd-operator/controllers/bpfd-agent/common.go

```diff
@@ -164,25 +164,50 @@ func (r *ReconcilerCommon) removeFinalizer(ctx context.Context, o client.Object,
 // false if the status was already set for the given bpfProgram, meaning reconciliation
 // may continue.
 func (r *ReconcilerCommon) updateStatus(ctx context.Context, prog *bpfdiov1alpha1.BpfProgram, cond bpfdiov1alpha1.BpfProgramConditionType) bool {
-	if prog.Status.Conditions != nil {
-		// Get most recent condition
-		recentIdx := len(prog.Status.Conditions) - 1
 
-		// If the most recent condition is the same as input, just return.
-		if prog.Status.Conditions[recentIdx].Type == string(cond) {
-			return false
-		} else {
-			// Remove the input condition from the list if it exists (may not exist)
-			// because the SetStatusCondition() doesn't append if it is already in the list.
-			meta.RemoveStatusCondition(&prog.Status.Conditions, string(cond))
+	r.Logger.V(1).Info("updateStatus()", "existing conds", prog.Status.Conditions, "new cond", cond)
+
+	if prog.Status.Conditions != nil {
+		r.Logger.V(1).Info("conditions != nil")
+		numConditions := len(prog.Status.Conditions)
+
+		switch {
+		case numConditions == 1:
+			{
+				if prog.Status.Conditions[0].Type == string(cond) {
+					r.Logger.V(1).Info("conditions match")
+					// No change, so just return false -- not updated
+					return false
+				} else {
+					// We're changing the condition, so delete this one.  The
+					// new condition will be added below.
+					r.Logger.V(1).Info("removing condition", "bpfProgram", prog.Name, "condition", prog.Status.Conditions[0].Type)
+					meta.RemoveStatusCondition(&prog.Status.Conditions, prog.Status.Conditions[0].Type)
+				}
+			}
+		case numConditions == 0:
+			{
+				// Fall through and add the new condition.
+			}
```

Comment:
nit remove this, just use `default:` and print 	`r.Logger.V(1).Info("Updating bpfProgram condition", "bpfProgram", prog.Name, "condition", cond.Condition().Type)`